packages/react/src/components/Heading/index.js

```diff
@@ -2,24 +2,24 @@ import React from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 
+import UI from "../UI";
 import { HEADING_LEVELS } from "../../constants";
 
-const levels = Object.keys(HEADING_LEVELS);
-
-const Heading = ({ as: Component, className, ...props }) => (
-  <Component
-    className={cx("Heading", `Heading--${Component}`, className)}
+const Heading = ({ as, className, ...props }) => (
+  <UI
+    type={`Heading${as}`}
```

Comment:
Not for right now but dropping the thought here: something we can consider is not using the prop `as` to pass data, that's used by StyledComponents (and other libraries) and having it here conflicts with it (last time I tried it).

Reply:
Yeah, `as` should be only for HTML tag 🤔 
Let's iterate on this later